service/src/main/kotlin/io/provenance/explorer/domain/extensions/CoinExtensions.kt

```diff
@@ -0,0 +1,23 @@
+package io.provenance.explorer.domain.extensions
+
+import java.math.BigDecimal
+import java.math.BigInteger
+import java.math.RoundingMode
+
+
+const val NHASH = "nhash"
+const val HASH = "hash"
+
+fun BigInteger.toHash(denom: String) =
+    (if (denom == NHASH)
+        Pair(this.toBigDecimal().divide(BigDecimal.valueOf(1000000000), 20, RoundingMode.HALF_UP), HASH)
+    else
+        Pair(this.toBigDecimal(), denom)
+    ).let { Pair(it.first.stripTrailingZeros().toPlainString(), it.second) }
```

Comment:
not sure, but removing all trailing zero's kinda munges the number of decimal places hash suppports(9 in this case)

```
fun main(){
    val b = BigDecimal("799999800199992800")
    println(b.divide(BigDecimal.valueOf(1000000000), 20, RoundingMode.HALF_UP).stripTrailingZeros())
```
gives: 799999800.1999928

`ROUND_UNNECESSARY` ..half up makes me uneasy and in principle it should be ROUND_UNNECESSARY
also i would use  UNNECESSARY(BigDecimal.ROUND_UNNECESSARY);

Reply:
If the hash was 0.100000000, would you rather see that or 0.1? Cuz Id rather see 0.1

Reply:
Everything else I agree with.

Reply:
would you rather see 0.1 for one tx, and then 0.011 for another and then 0.00000001 for another..personally i would like to the number of decimal places supported, but idk.. i guess keep it as is 🤷